### ISSUE TITLE
Finetuning Cellpose-SAM

### ISSUE THREAD

**Issue by licmn** [MouseLand/cellpose#1353]
I have 17 spatial transcriptomics samples (1 sample shown) with paired H&E images. I am looking to predict cell types from H&E, and for that, I finetuned Cellpose-SAM model for predicting 13 classes. 

<img width="473" height="611" alt="Image" src="https://github.com/user-attachments/assets/cae8b067-9878-42c1-b120-5518aaad4863" />

<img width="394" height="462" alt="Image" src="https://github.com/user-attachments/assets/deba42ec-c4ac-4116-ac35-fd7dac34562c" />

I followed Cellpose semantic segmentation code ([https://github.com/MouseLand/cellpose/blob/e3879a1cc58d4aa313d50977bb9b31ab11f89a2e/paper/cpsam/semantic.py](https://github.com/MouseLand/cellpose/blob/main/paper/cpsam/semantic.py)), only changing there the number of labels. I used their hyperparameters as well:

```
# Hyperparameters:
rdrop=0.4 # In vit_sam.Transformer initialization
learning_rate = 5e-5 
weight_decay = 0.1 
batch_size = 8 
n_epochs = 500
bsize = 256
rescale = False 
scale_range = 0.5
```

Also, I’ve read about the mean cell diameter parameter for cellpose, but have not used it. Here is the loss curve from finetuning. I can see that past epoch 150-200 the model starts to overfit on the training data without improving performance on the withheld test samples. I would truly appreciate any advice or feedback on how to fine-tune the model better. What would you try changing first in the finetuning process to improve training?

Here is the training curve

<img width="1232" height="726" alt="Image" src="https://github.com/user-attachments/assets/67e5b87b-55ac-47d2-9785-a8b13e50dd51" />

Thanks a lot for your help!
Luisa

**Comment by Jiadalee** [comment]
@licmn pls lower down your n_epoch to 50-30 and  batch_size to 2-3. You might also remove the rdrop and scale_range params.

**Comment by mrariden** [comment]
Hi @licmn, I think you should evaluate how well your models performs on train/test data using accuracy/recall or other metrics to see performance. That will give you more insight into how bad the overfitting is. Adding the semantic segmentation makes the training objective more complex because the network has to learn both the cellpose target and type simultaneously.

Depending on your task/data it may be worth using cellpose to get segmentations and then train a separate classifier. I say this because semantic segmentation isn't the typical use case for cellpose. 

Still, in general, decreasing the learning rate, increasing batch size, and using rdrop will help with regularization which should mitigate overfitting

**Comment by carsen-stringer** [comment]
You could also try the newer DINO models that can use a larger tile size (bsize=384 or bsize=512), if that is necessary to determine the class. It might also be the case that you are already getting the best performance, unless you can clearly see mistakes that you can tell by eye very clearly. Let us know if you have more questions

**Comment by Jiadalee** [comment]
> You could also try the newer DINO models that can use a larger tile size (bsize=384 or bsize=512), if that is necessary to determine the class. It might also be the case that you are already getting the best performance, unless you can clearly see mistakes that you can tell by eye very clearly. Let us know if you have more questions

@carsen-stringer Congrats to the new release of the newest DINO models. I didn't notice that until your response.
